watertap/property_models/anaerobic_digestion/modified_adm1_reactions.py

```diff
@@ -0,0 +1,2604 @@
+        )
+
+        def concentration_of_pro_rule(self):
+            return self.conc_mass_pro == self.params.K_a_pro * self.conc_mass_comp_ref[
+                "S_pro"
+            ] / (self.params.K_a_pro + self.S_H)
+
+        self.concentration_of_pro = pyo.Constraint(
+            rule=concentration_of_pro_rule,
+            doc="constraint concentration of pro-",
+        )
+
+        def concentration_of_ac_rule(self):
+            return self.conc_mass_ac == self.params.K_a_ac * self.conc_mass_comp_ref[
+                "S_ac"
+            ] / (self.params.K_a_ac + self.S_H)
+
+        self.concentration_of_ac = pyo.Constraint(
+            rule=concentration_of_ac_rule,
+            doc="constraint concentration of ac-",
+        )
+
+        def concentration_of_hco3_rule(self):
+            return self.conc_mol_hco3 == self.K_a_co2 * (
+                self.conc_mass_comp_ref["S_IC"] / (12 * pyo.units.kg / pyo.units.kmol)
+            ) / (self.K_a_co2 + self.S_H)
+
+        self.concentration_of_hco3 = pyo.Constraint(
+            rule=concentration_of_hco3_rule,
+            doc="constraint concentration of hco3",
+        )
+
+        def concentration_of_nh3_rule(self):
+            return self.conc_mol_nh3 == self.K_a_IN * (
+                self.conc_mass_comp_ref["S_IN"] / (14 * pyo.units.kg / pyo.units.kmol)
+            ) / (self.K_a_IN + self.S_H)
+
+        self.concentration_of_nh3 = pyo.Constraint(
+            rule=concentration_of_nh3_rule,
+            doc="constraint concentration of nh3",
+        )
+
+        # TO DO: use correct conversion number
+        def concentration_of_co2_rule(self):
+            return (
+                self.conc_mol_co2
+                == self.conc_mass_comp_ref["S_IC"]
+                / (12 * pyo.units.kg / pyo.units.kmol)
+                - self.conc_mol_hco3
+            )
+
+        self.concentration_of_co2 = pyo.Constraint(
+            rule=concentration_of_co2_rule,
+            doc="constraint concentration of co2",
+        )
+
+        def concentration_of_nh4_rule(self):
+            return (
+                self.conc_mol_nh4
+                == self.conc_mass_comp_ref["S_IN"]
+                / (14 * pyo.units.kg / pyo.units.kmol)
+                - self.conc_mol_nh3
+            )
+
+        self.concentration_of_nh4 = pyo.Constraint(
+            rule=concentration_of_nh4_rule,
+            doc="constraint concentration of nh4",
+        )
+
+        def concentration_of_Mg_rule(self):
+            return self.conc_mol_Mg == self.conc_mass_comp_ref["X_PP"] / (
+                300.41 * pyo.units.kg / pyo.units.kmol
+            )
+
+        self.concentration_of_Mg = pyo.Constraint(
+            rule=concentration_of_Mg_rule,
+            doc="constraint concentration of Mg",
+        )
+
+        def concentration_of_K_rule(self):
+            return self.conc_mol_Mg == self.conc_mass_comp_ref["X_PP"] / (
+                300.41 * pyo.units.kg / pyo.units.kmol
```

Comment:
This seems to be a copy of the previous constraint.

Reply:
Thx for catching this - the expressions are equivalent, but I did forget to replace `conc_mol_Mg` with `conc_mol_K`

Reply:
What molecule does that molecular weight correspond to? I figure X_PP represents polyphosphates but unsure what the MW assumption specifically ties to.

Reply:
![image](https://user-images.githubusercontent.com/96305519/228537960-26834b21-6cdb-4b1f-b52d-b972106a72dc.png)